src/container-mapping.ts

```diff
@@ -215,4 +224,70 @@ export class ContainerMapping implements IMicrosoftSecurityDevOps {
             req.end();
         });
     }
+
+    /**
+     * Queries Defender for DevOps to determine if the caller is onboarded for container mapping.
+     * @param retryCount the number of time to retry
+     * @param bearerToken the GitHub-generated OIDC token
+     * @returns a boolean Promise to indicate if the report was sent successfully or not
+     */
+    private async checkCallerIsCustomer(bearerToken: string, retryCount: number = 0): Promise<boolean> {
+        return await this._checkCallerIsCustomer(bearerToken)
+        .then(async (statusCode) => {
+            if (statusCode == 200) { // Status 'OK' means the caller is an onboarded customer.
+                return true;
+            } else if (statusCode == 403) {// Status 'Forbidden' means caller is not a customer.
+                return false;
+            } else {
+                core.debug(`Unexpected status code: ${statusCode}`);
+                if (retryCount == 0) {
+                    return false;
+                } else {
+                    core.info(`Retrying API call.\nRetry count: ${retryCount}`);
+                    retryCount--;
+                    return await this.checkCallerIsCustomer(bearerToken, retryCount);
+                }
+            }
+        })
+        .catch(async (error) => {
+            if (retryCount == 0) {
+                return false;
+            } else {
+                core.info(`Retrying checkCallerIsCustomer call due to error: ${error}.\nRetry count: ${retryCount}`);
+                retryCount--;
+                return await this.checkCallerIsCustomer(bearerToken, retryCount);
+            }
+        });
+    }
+
+    private async _checkCallerIsCustomer(bearerToken: string): Promise<number> {
+        return new Promise(async (resolve, reject) => {
+            let url: string = "https://dfdinfra-afdendpoint-prod-d5fqbucbg7fue0cf.z01.azurefd.net/github/v1/auth-push/GetScanContext?context=authOnly";
+            let options = {
+                method: 'GET',
+                timeout: 2500,
+                headers: {
+                    'Content-Type': 'application/json',
+                    'Authorization': 'Bearer ' + bearerToken,
+                }
+            };
+            core.debug(`${options['method'].toUpperCase()} ${url}`);
+
+            const req = https.request(url, options, (res) => {
+
+                res.on('end', () => {
+                    resolve(res.statusCode);
+                });
+                res.on('data', function(d) {

```

Comment:
Is this needed?

Reply:
Unfortunately. Even though the function doesn't do anything, registering any listener on the response requires the caller to register a listener on the 'data' event for the stream to be closed.

Reply:
Ahh good to know, didnt know this part.